### PR TITLE
Move labels to the same device as logits for Whisper

### PR DESCRIPTION
**Pull request by oscar-garzon**
# What does this PR do?

Fixes issue #22561 by moving labels to the same device as logits for `Whisper` model.

@sgugger Could you please review?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!
